src/Storages/VirtualColumnUtils.cpp

```diff
@@ -389,19 +389,19 @@ void addRequestedPathAndFileVirtualsToChunk(
     {
         if (virtual_column.name == "_path")
         {
-            chunk.addColumn(virtual_column.type->createColumnConst(chunk.getNumRows(), path));
+            chunk.addColumn(virtual_column.type->createColumnConst(chunk.getNumRows(), path)->convertToFullColumnIfConst());
```

Comment:
If we would agree to proceed with option (1), then the comment is needed which describes this behaviour + we need to check other places where we inject virtual columns.